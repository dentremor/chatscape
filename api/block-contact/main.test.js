const mockQuery = jest.fn();
const mockUpdate = jest.fn();

jest.mock("aws-sdk", () => {
  return {
    DynamoDB: {
      DocumentClient: jest.fn().mockImplementation(() => ({
        query: mockQuery,
        update: mockUpdate,
      })),
    },
  };
});

const { handler } = require("./main");

describe("Block contact lambda", () => {
  it("sets the blocked boolean after a user has been blocked", async () => {
    mockQuery.mockReturnValueOnce({
      promise: jest.fn().mockResolvedValueOnce({
        Items: [
          { email: "test@example.com", id: "1234" },
          { email: "another@example.com", id: "5678" },
        ],
      }),
    });

    mockUpdate.mockReturnValueOnce({
      promise: jest.fn().mockResolvedValueOnce(),
    });

    const rv = await handler({
      httpMethod: "POST",
      body: JSON.stringify({
        namespace: "test",
        email: "test@example.com",
      }),
    });

    expect(mockUpdate).toHaveBeenCalledWith({
      TableName: undefined,
      Key: {
        id: "1234",
      },
      UpdateExpression: "set blocked = :blockedVal",
      ExpressionAttributeValues: {
        ":blockedVal": true,
      },
    });

    expect(rv.statusCode).toBe(200);
    expect(rv).toMatchSnapshot();
  });
});
