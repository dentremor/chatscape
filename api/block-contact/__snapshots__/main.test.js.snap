// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block contact lambda sets the blocked boolean after a user has been blocked 1`] = `
{
  "body": "{"email":"test@example.com","id":"1234","blocked":true}",
  "headers": {
    "Access-Control-Allow-Origin": undefined,
  },
  "statusCode": 200,
}
`;

exports[`Block contact lambda should handle contact not found 1`] = `
{
  "body": ""contact not found"",
  "headers": {
    "Access-Control-Allow-Origin": undefined,
  },
  "statusCode": 404,
}
`;

exports[`Block contact lambda should handle invalid request body 1`] = `
{
  "body": ""invalid request body"",
  "headers": {
    "Access-Control-Allow-Origin": undefined,
  },
  "statusCode": 400,
}
`;
